src/mon/ConfigMonitor.cc

```diff
@@ -333,30 +334,34 @@ bool ConfigMonitor::preprocess_command(MonOpRequestRef op)
 	goto reply;
       }
       if (opt->has_flag(Option::FLAG_NO_MON_UPDATE)) {
-	// handle special options
-	if (name == "fsid") {
-	  odata.append(stringify(mon.monmap->get_fsid()));
-	  odata.append("\n");
-	  goto reply;
-	}
-	err = -EINVAL;
-	ss << name << " is special and cannot be stored by the mon";
-	goto reply;
+        // handle special options
+        if (name == "fsid") {
+          value = stringify(mon.monmap->get_fsid());
+        } else {
+          err = -EINVAL;
+          ss << name << " is special and cannot be stored by the mon";
+          goto reply;
+        }
       }
       // get a single value
       auto p = config.find(name);
       if (p != config.end()) {
-	odata.append(p->second);
-	odata.append("\n");
-	goto reply;
+        value = p->second;
+      } else if (!entity.is_client() &&
+              opt->daemon_value != Option::value_t{} ) {
+        value = opt->daemon_value;
+      } else {
+        value = opt->value;
       }
-      if (!entity.is_client() &&
-	  opt->daemon_value != Option::value_t{}) {
-	odata.append(Option::to_str(opt->daemon_value));
+      if (f) {
+        f->open_object_section("option");
+        f->dump_string(name, Option::to_str(value));
+        f->close_section();
```

Comment:
What about:

```suggestion
        f->open_object_section("config");
        f->dump_string(name, Option::to_str(value));
        f->dump_object("option", opt);
        opt->dump_value("value", value, f);
        f->close_section();
```

I don't see a downside to supporting both approaches. The extra information is useful, in particular `opt->dump_value("value", value, f);` which preserves the type.

Reply:
We've discussed a similar case [here](https://github.com/ceph/ceph/pull/52882#discussion_r1288945894). The reasoning is the same: I don't think this is a good time to introduce this - IMO the correct one - approach. All other configuration is dumped as string.

Reply:
OK sorry, I see you are suggesting a different response schema. Maybe this will be accepted, but that is back to the open question about what the schema should be. As of now, I'm leaning towards the `{"name":"value"}` as the one that matches best with the other commands

Reply:
https://github.com/ceph/ceph/pull/52882#issuecomment-1671038579

Reply:
I'm recommending you keep the `{"name":"value"}` response format while adding additional information which is available.

Reply:
OK. Let me see if I understand correctly, are you suggesting that we respond with 

```
{
    "auth_debug": "true",
    "value": true
}
```